### DISCUSSION TITLE
401 Unauthorized

### DISCUSSION THREAD

**Discussion by yairShiloach** [chatwoot/chatwoot#2149] [General]
Hi,
I can't get or post any information via api - I always get the error:

{
     "errors": [
         "You need to sign in or sign up before continuing."
     ]
}

I did everything according to the docs, using a parameter in the header called "api_access_token"

this the url to get conversion:
https://www.example.com/api/v1/accounts/1/conversations/1

**Comment by yairShiloach** [comment]
"Filter chain halted as :authenticate_user! rendered or redirected"

**Reply by sojan-official** [reply]
could you give more information about your chatwoot environment ?
is it behind an Nginx frontend proxy server?

**Reply by yairShiloach** [reply]
Installed using a caprover on DigitalOcean
last version: 1.15.1
In addition to this issue, the SMTP setting does not work and does not send the message... 
Maybe it's related to the same problem.

Maybe I should have set the Environmental Variables before setting up the app for the first time?

**Comment by sojan-official** [comment]
@yairShiloach  nginx by default strip of headers with ` _` . head over to the Nginx configuration option in caprover under the chatwoot web and add the following directive. 

```
  # Nginx strips out underscore in headers by default
  # Chatwoot relies on underscore in headers for API
  # Make sure that the config is set to on.
  underscores_in_headers on;
  ```

**Reply by yairShiloach** [reply]
thanks!

**Reply by gvargas9** [reply]
Thanks!  I was going crazy!

**Reply by jniemand** [reply]
This solved it, thanks!
